packages/frontend-2/lib/common/composables/dialog.ts

```diff
@@ -0,0 +1,228 @@
+import { LogicError } from '@speckle/ui-components'
+import { upperFirst, values } from 'lodash-es'
+import type { ModelRef, Ref } from 'vue'
+
+export type DialogStep<ID extends string = string> = {
```

Comment:
This is really useful! There are some other modals that can need some attention and can implement this, I will update those